### DISCUSSION TITLE
Is it possible to safely share a single hot wallet between multiple stores?

### DISCUSSION THREAD

**Discussion by Kixunil** [btcpayserver/btcpayserver#2886] [Technical Support]
I'd like to share a hot wallet between multiple stores **with PayJoin enabled** but I don't know if/how it's possible to do it safely.
The obvious solution seems to be to enter same seed on second store but is NBXplorer/BTCPayServer written such that multiple concurrent PayJoins don't attempt to double spend inputs or cause other problems?

If it's safe then it'd be nice to explicitly say so in doc. If it's not I would like this to be considered feature request.

Note that the main reason why I need this is to have a crowdfunding store with email address question and a donation button without such question. So an alternative would be for email question to be app-based rather than store-based. Frankly this might be even better.

**Comment by ghost** [comment]
> So an alternative would be for email question to be app-based rather than store-based.

I'd like this too.

**Reply by bolatovumar** [reply]
I think this makes sense as you can have two POS apps under the same store and one of those store requires email while the other doesn't. It's kind of weird that you have to create another store just so that you can create an app which has a different setting for requiring email on invoice.

**Reply by bolatovumar** [reply]
@Kixunil @MaxHillebrand While implementing this I realized having both a store-level and an app level option for requiring an email address doesn't really make sense as the app level option will always override the store level one (since it's either on or off). 

Also, for backwards compatibility with existing stores we have to keep the store-level option around as we don't want to have existing stores which have this turned on to suddenly have it off because it's turned off by default at app level.

In any case I made a PR with WIP code here (it works but it's not really polished so I opened and closed the PR right away for now, if we decide to proceed further with this I can reopen it): https://github.com/btcpayserver/btcpayserver/pull/3008

Thoughts on this @pavlenex?

**Reply by Kixunil** [reply]
Maybe it could have three states? On, Off, Inherit from store. Existing instances would use "Inherit from store".

**Reply by bolatovumar** [reply]
@Kixunil good idea. Updated my PR and re-opened it here: https://github.com/btcpayserver/btcpayserver/pull/3013

**Reply by Kixunil** [reply]
Thanks a lot!
